HandDetector.py

```diff
@@ -0,0 +1,125 @@
+
+        self.input_details = self.interpreter.get_input_details()
+        self.output_details = self.interpreter.get_output_details()
+
+        self.imageTargetSize = (256, 256)
+    
+    def preProcessImage(self, image):
+        resized = cv2.resize(image, self.imageTargetSize)
+        return resized
+    
+    @staticmethod
+    def _sigm(x):
+        return 1 / (1 + np.exp(-x) )
+
+    def resizeOriginal(self, keypoints, center, original_width, original_height):
+        kp_orig = np.copy(keypoints)
+        cnt_orig = np.copy(center)
+
+        (w, h) = self.imageTargetSize
+
+        kp_orig[:,0] *= original_width/w
+        kp_orig[:,1] *= original_height/h
+
+        cnt_orig[0] *= original_width/w
+        cnt_orig[1] *= original_height/h
+
+        return (kp_orig, cnt_orig)
+
+    def getCenter(self, keypoints):
+        return np.mean(keypoints, axis= 0)
+
+    @staticmethod
+    def getBBox(keypoints, center, side_multiplier = 3):
+        maxes = np.amax(keypoints, axis = 0)
+        mins = np.amin(keypoints, axis = 0)
+        sides = np.abs(maxes - mins)
+        max_side = np.amax(sides)
+        side = side_multiplier * max_side
+        x = center[0] - side/2
+        y = center[1] - side/2
+        return (x, y, side, side)
+
+
+    def __call__(self, image):
+        preProcessed = self.preProcessImage(image)
+        input_img = tf.expand_dims(tf.image.convert_image_dtype(preProcessed, dtype=tf.float32), 0)
+
+        self.interpreter.set_tensor(self.input_details[0]['index'], input_img)
+
+        self.interpreter.invoke()
+
+        out_reg = self.interpreter.get_tensor(self.output_details[0]['index'])[0]
+        out_clf = self.interpreter.get_tensor(self.output_details[1]['index'])[0,:,0]
+
+        detecion_mask = self._sigm(out_clf) > 0.7
+
+        candidate_detect = out_reg[detecion_mask]
+
+        candidate_anchors = self.anchors[detecion_mask]
+
+        if candidate_detect.shape[0] != 0:
+
+            max_idx = np.argmax(candidate_detect[:, 3])
+
+            dx,dy,w,h = candidate_detect[max_idx, :4]
+            center_wo_offst = candidate_anchors[max_idx,:2] * 256
+
+            keypoints = center_wo_offst + candidate_detect[max_idx,4:].reshape(-1,2)
+            
+            center = self.getCenter(keypoints)
+
+            (res_kp, res_cnt) = self.resizeOriginal(keypoints, center, frame.shape[1], frame.shape[0])
+
+            return (res_kp, res_cnt)
+        else:
+            return (None, None)
+        
+if __name__ == '__main__':
+    curr_dir = os.getcwd()
+    print(curr_dir)
+
+    palm_model_path = curr_dir + "/models/palm_detection.tflite"
+    anchors_path = curr_dir + "/models/anchors.csv"
+
+    detector = HandDetector(palm_model_path, anchors_path)
+
+    capture = cv2.VideoCapture(0)
+
+    while True:
+
+        ret, frame = capture.read()
+        image = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+
+        keypoints, center = detector(image)
+
+        if keypoints is not None:
```

Comment:
z jakiegoś konkretnego powodu jawnie porównujesz z None? Jak nie, to lepiej zamienić na `if keypoints:`

Reply:
None jest zwracany jeśli model nie wykrył dłoni :P

Tutaj jest, że porównując z None  lepiej korzystać z operatora 'is' https://stackoverflow.com/questions/3965104/not-none-test-in-python